lib/libc/Kconfig

```diff
@@ -138,6 +138,21 @@ config MINIMAL_LIBC_LL_PRINTF
 	  Build with long long printf enabled. This will increase the size of
 	  the image.
 
+config MINIMAL_LIBC_GLOBALS
+	bool "Enables functions require global variables"
+	default y
+	help
+	  Enable the minimal libc's implementation of functions which
+	  implicitly access global/static data, e.g. rand(). Such data
+	  must be put into a memory partition if CONFIG_USERSPACE=y,
+	  and disabling this option may save an entry for application
+	  defining many custom partitions.
+
+	  Say 'N' here if you want to save a memory partition by
+	  disabling the following functions:
+	    - rand
+	    - srand
+
```

Comment:
Noting that we already have the configs to selectively enable/disable different features of the minimal libc, I think it would be better to add a config to enable the "rand feature" and make that select `NEED_LIBC_MEM_PARTITION` so that the globals can be used for userspace.

For example:
```
config MINIMAL_LIBC_RAND
	bool "Enable minimal libc rand implementation"
	select NEED_LIBC_MEM_PARTITION
...
```

Reply:
p.s. I see that the PR originally added `MINIMAL_LIBC_RAND`, but later changed to `MINIMAL_LIBC_GLOBALS` because memory protection tests were failing due to the libc memory partition being added and there being insufficient number of MPU regions being available for low end Cortex-M cores.

FWIW, I do not think it warrants reworking the config scheme just to get around that.

The current minimal libc implementation explicitly does not make use of the globals due to the high cost associated with it (i.e. additional memory partition for libc). Noting that, I think it is better to just make any minimal libc features that `select NEED_LIBC_MEM_PARTITION` not enabled by default, so users can enable those features if and only if they need them.

Reply:
Reverted back to `CONFIG_MINIMAL_LIBC_RAND` instead of `CONFIG_MINIMAL_LIBC_GLOBALS`.

Reply:
And also `CONFIG_MINIMAL_LIBC_RAND` now selects `CONFIG_NEED_LIBC_MEM_PARTITION`.